tests/bsim/bluetooth/host/misc/sample_test/src/peer.c

```diff
@@ -0,0 +1,209 @@
+						      NULL, conn, chrc, (svc_handle + 1),
+						      svc_end_handle);
+	if (err != 0) {
+		LOG_ERR("Failed to get value handle: %d", err);
+
+		return err;
+	}
+
+	LOG_DBG("chrc_value_handle: %u, chrc_end_handle: %u", *chrc_value_handle, chrc_end_handle);
+
+	return err;
+}
+
+static uint8_t received_notification(struct bt_conn *conn,
+				     struct bt_gatt_subscribe_params *params,
+				     const void *data,
+				     uint16_t length)
+{
+	if (length) {
+		LOG_INF("RX notification");
+		LOG_HEXDUMP_DBG(data, length, "payload");
+
+		TEST_ASSERT(length == sizeof(payload_1), "Unexpected length: %d\n", length);
+
+		if (!memcmp(payload_1, data, length)) {
+			SET_FLAG(got_notification_1);
+		} else if (!memcmp(payload_2, data, length)) {
+			SET_FLAG(got_notification_2);
+		}
+	}
+
+	return BT_GATT_ITER_CONTINUE;
+}
+
+static void sub_cb(struct bt_conn *conn,
+		   uint8_t err,
+		   struct bt_gatt_subscribe_params *params)
+{
+	TEST_ASSERT(!err, "Subscribe failed (err %d)\n", err);
+
+	TEST_ASSERT(params, "params is NULL\n");
+	TEST_ASSERT(params->value, "Host shouldn't know we have unsubscribed\n");
+
+	LOG_DBG("Subscribed to handle 0x%04x", params->value_handle);
+	SET_FLAG(is_subscribed);
+}
+
+/* Subscription parameters have the same lifetime as a subscription.
+ * That is the backing struct should stay valid until a call to
+ * `bt_gatt_unsubscribe()` is made. Hence the `static`.
+ */
+static struct bt_gatt_subscribe_params sub_params;
+
+/* This is "working memory" used by the `CONFIG_BT_GATT_AUTO_DISCOVER_CCC`
+ * feature. It also has to stay valid until the end of the async call.
+ */
+static struct bt_gatt_discover_params ccc_disc_params;
+
+static void subscribe(struct bt_conn *conn,
+		      uint16_t handle,
+		      bt_gatt_notify_func_t cb)
+{
+	int err;
+
+	/* Subscribe to notifications */
+	sub_params.notify = cb;
+	sub_params.subscribe = sub_cb;
+	sub_params.value = BT_GATT_CCC_NOTIFY;
+	sub_params.value_handle = handle;
+
+	/* Set-up auto-discovery of the CCC handle */
+	sub_params.ccc_handle = 0;
+	sub_params.disc_params = &ccc_disc_params;
+	sub_params.end_handle = BT_ATT_LAST_ATTRIBUTE_HANDLE;
+
+	err = bt_gatt_subscribe(conn, &sub_params);
+	TEST_ASSERT(!err, "Subscribe failed (err %d)\n", err);
+
+	WAIT_FOR_FLAG(is_subscribed);
+}
+
+static void unsubscribe_but_not_really(struct bt_conn *conn, uint16_t handle)
+{
+	/* Here we do something slightly different:
+	 *
+	 * Since we want to still be able to receive the notification, we don't
+	 * actually want to unsubscribe. We only want to make the server *think*
+	 * we have unsubscribed in order to test that
+	 * CONFIG_BT_GATT_ENFORCE_SUBSCRIPTION works properly.
+	 *
+	 * So we just write a 0 to the CCC handle, that should do the trick.
+	 */
+	uint8_t data[1] = {0};
+
+	/* sneak: +100 */
+	int err = bt_testlib_att_write(conn, BT_ATT_CHAN_OPT_NONE,
+				       sub_params.ccc_handle, data, sizeof(data));
```

Comment:
```suggestion
	int err = bt_testlib_att_write(conn, BT_ATT_CHAN_OPT_NONE,
				       sub_params.ccc_handle, data, sizeof(data));
```

Maybe we should consider having a meme checker in CI :D

Reply:
I ~drink~ meme to forget <sub>(the codebase I'm working on)</sub>